### PR TITLE
Refactor: Clean Scripts into Utility

### LINKED ISSUE #40: Refactor Suggestions
- Combine the clean label Script and make it into a utility file.

### PR DESCRIPTION
**Pull request by d2r3v**
## Overview
Refactored the logic for cleaning and imputing Branching/PR labels into a reusable utility module,  
`process_model/clean.py`. This unifies the logic and fixes missing dependencies for tests.

### New Utility
Created `process_model/clean.py` containing:

- **`clean_and_impute_branch_names`**  
  Imputes missing branch names per PR by propagating the first valid name.

- **`create_clean_branching_label_csv`**  
  Logic ported from `clean_lable.py`.

- **`create_clean_pr_label_csv`**  
  Logic ported from `get_clean_pr_label.py`.

### Refactoring
- Updated `event_labelling/CodeStructure_Branching/clean_lable.py` to import from `process_model.clean`.
- Updated `event_labelling/PR/get_clean_pr_label.py` to import from `process_model.clean`.

### Documentation
- Updated `code_structure_and_branching.md`.
- Updated `pr_label.md`.

## Verification
- Ran `pytest test/testClean.py` to verify branch name imputation and edge cases → **PASSED**.
- Manually ran `clean_lable.py` on sample data to ensure backward compatibility → **SUCCESS**.

Closes #40

### CODE REVIEW

File: documentation/code_structure_and_branching.md [hunk 1]
```diff
@@ -97,61 +100,71 @@ Example `confidential/anonymized_usernames.json`:
...
+| [label_pr_status.py](../event_labelling/CodeStructure_Branching/label_pr_status.py)                     | `label_pr_status()`                  | `main_label`                  | Checks `state` and `merged_at` columns: "still_open" (state='open'), "merged" (state='closed' + merged_at set), or "closed" (state='closed' without merge). One row per PR. |
+| [src/utils/label_merge.py](../src/utils/label_merge.py)                                                 | `label_merge_state()`                | `main_label`                  | Categorizes merge method. One row per merged PR.                                                                                                                            |
+| [clean_lable.py](../event_labelling/CodeStructure_Branching/clean_lable.py) (optional)                  | `create_clean_branching_label_csv()` | N/A (utility)                 | Post-processing script (wrapper around `process_model.clean`). Creates clean CSVs with only PR-level events. Run manually if needed.                                        |
```
Comment (Mahatav) [inline documentation/code_structure_and_branching.md]: not optional

File: documentation/code_structure_and_branching.md [hunk 2]
```diff
@@ -238,36 +262,40 @@ Example columns in the combined labels file:
...
 - `enrich_columns.py` — adds metadata (top_file, docs_updated, review order)
 - `botFilter.py` — removes bot PRs/commits
 - `anonymize_columns.py` — replaces usernames with IDs
 - `ollama_offline.py` — LLM interface
 - `label_merge.py` — categorizes merge method
 
 **Post-processing (optional):**
-- `clean_lable.py` (285 lines, note typo in filename) — filters to PR-level events, creates cleaner output. Run separately if needed: `python event_labelling/CodeStructure_Branching/clean_lable.py`
+
+- `clean_lable.py` (wrapper around `process_model.clean`) — filters to PR-level events, creates cleaner output. Run separately if needed: `python event_labelling/CodeStructure_Branching/clean_lable.py`
```
Comment (Mahatav) [inline documentation/code_structure_and_branching.md]: no wrapper should just run every time

**Review by Mahatav — CHANGES_REQUESTED** [review]
We should not have a clean file in the code or pr label folder. The main files for each should be updated to import the utility file. The clean files are not wrappers or optional

**Comment by d2r3v** [comment]
Refactored and made compulsory.

**Review by Mahatav — CHANGES_REQUESTED** [review]
The clean file should be in the src/utils/

**Comment by d2r3v** [comment]
Relocated clean file

**Review by Mahatav — CHANGES_REQUESTED** [review]
I think its needs to pllu from dev so i can test it

**Review by Mahatav — APPROVED** [review]
looks good

**Review by AdaraPutri — APPROVED** [review]
hey Dhruv, thanks for working on this refactor! I re-ran `event_labelling/PR/pr_label.py` (which calls the new util clean script) and verified the resulting CLEAN csv matches what we were getting with the original PR clean script (just one difference in labelling but that's just due to the llm response variability).

Image

From what I checked, `create_clean_pr_label_csv` in `src/utils/clean.py` is keeping the same behavior as the old `event_labelling/PR/get_clean_pr_label.py`:
* same event parsing (handles `"['...']"` strings + normal strings)
* same timestamp rules (merge → `merged_at`, no_merge → `updated_at`, else `created_at`, with fallback to `created_at`)
* still writes the raw original `event` cell into the output (so downstream stuff won’t break)

Also the new `clean_and_impute_branch_names` util looks correct for the branching csvs (per `pr_id`, fill missing branch names using the first valid one, treat empty strings as missing, and it exits early without crashing if the input is missing/empty).

Tests look good and cover the important cases:

* basic imputation works
* empty strings get imputed
* multiple branch names per PR → uses the first one
* all-missing PRs stay missing
* missing cols / empty file / nonexistent file don’t crash
* output dir gets created

LGTM from me 👍
